test/verify/check-storage-anaconda

```
@@ -0,0 +1,201 @@
+
+    def testBasic(self):
+        m = self.machine
+        b = self.browser
+
+        disk = self.add_ram_disk()
+
+        anaconda_config = {
+            "mount_point_prefix": "/sysroot",
+            "available_devices": [disk],
+        }
+
+        self.login_and_go("/storage")
+        b.call_js_func("window.localStorage.setItem", "cockpit_anaconda", json.dumps(anaconda_config))
+        b.reload()
+        b.enter_page("/storage")
+
+        # There should be only one row, for our disk
+        b.wait(lambda: b.call_js_func('ph_count', self.card("Storage") + " tbody tr") == 1)
+        b.wait_text(self.card_row_col("Storage", 1, 3), "Unformatted data")
+        b.wait_not_present(self.card_row("Storage", location="/"))
+
+        # Create a volume group with a logical volume
+        self.click_devices_dropdown("Create LVM2 volume group")
+        self.dialog_wait_open()
+        b.wait(lambda: b.call_js_func('ph_count', "#dialog .select-space-name") == 1)
+        self.dialog_set_val("disks", {disk: True})
+        self.dialog_apply()
+        self.dialog_wait_close()
+        self.click_dropdown(self.card_row("Storage", name="vgroup0"), "Create new logical volume")
+        self.dialog({})
+
+        # Create an encrypted filesystem
+        self.click_dropdown(self.card_row("Storage", name="lvol0"), "Format")
+        self.dialog_wait_open()
+        self.dialog_wait_val("at_boot", "local")
+        self.dialog_set_val("mount_point", "/")
+        self.dialog_set_val("type", "ext4")
+        self.dialog_set_val("crypto", self.default_crypto_type)
+        self.dialog_set_val("passphrase", "vainu-reku-toma-rolle-kaja")
+        self.dialog_set_val("passphrase2", "vainu-reku-toma-rolle-kaja")
+        self.dialog_apply()
+        self.dialog_wait_close()
+
+        self.assertNotIn("noauto", m.execute("findmnt --fstab -n -o OPTIONS /sysroot"))
+        self.assertNotIn("nofail", m.execute("findmnt --fstab -n -o OPTIONS /sysroot"))
```

`findmnt` will fail if the mount doesn't exist, but it would still feel a bit better to also have some positive test here. Maybe that it contains `lvol0` and `defaults` or `auto`?

---

We have a positive test further down, is that enough?

---

There isn't really anything interesting in options...

---

Right, but it can also show SOURCE ?

---

> We have a positive test further down, is that enough?

I guess.. I don't want to dwell on this, I leave this up to you. TBH, my gut feeling is that this is by far not the end of the whole Anaconda integration story, we need to have some much more conceptual discussions. Hence, I'll almost surely approve your next force-push :grin:

---

> TBH, my gut feeling is that this is by far not the end of the whole Anaconda integration story,

Of course not. :-)  The current goal is to get something into Cockpit main that Anaconda can test against.

---

> I guess.. I don't want to dwell on this, I leave this up to you.

Ok, let's leave it as it is. As you say, findmnt will fail noisily when the expected /sysroot entry is not there.